src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js

```diff
@@ -1,16 +1,60 @@
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import {
   faDownload
 } from '@fortawesome/free-solid-svg-icons';
 
 import Modal from 'react-bootstrap/Modal';
 import Button from 'react-bootstrap/Button';
+import styles from './index.module.css';
+
 import download from 'downloadjs';
 
-const handleClick = (url, fileName) => {
-  download(url, fileName, 'application/octet-stream');
+const DownloadItem = ({ name, fileName, url, type }) => {
+  const [ isDownloading, setIsDownloading ] = useState(false);
+  const [ progress, setProgress ] = useState();
+
+  const handleProgress = (e) => {
+    if (e.lengthComputable) {
+      setProgress(Math.round((e.loaded / e.total) * 100));
+    }
+  };
+
+  const handleClick = () => {
+    const x = new XMLHttpRequest();
+    x.addEventListener('loadstart', setIsDownloading(true));
+    x.addEventListener('progress', handleProgress);
+    x.open('GET', url, true);
+    x.responseType = 'blob';
+    x.onload = function () {
+      download(x.response, fileName, type);
+      setIsDownloading(false);
+    };
+    x.send();
+  };
+
+  const getStatus = () => {
+    if (isDownloading) {
+      return (
+        <span className={ styles.status }>Downloading... { progress }%</span>
```

Comment:
as mentioned above, would prefer for it to look like:
```suggestion
        <span style={{ css thing }}>Downloading... { progress }%</span>
```